### ISSUE TITLE
[pub] : Dependabot fails after upgrading flutter 

### ISSUE THREAD

**Issue by SiddheshKukade** [dependabot/dependabot-core#6867]
please take look at these questions:
I've explained everything in detail
https://stackoverflow.com/questions/75751974/dependabot-version-fails-due-to-outdated-flutter-version

**Comment by jeffwidman** [comment]
related:
* https://github.com/PalisadoesFoundation/talawa/issues/1643

The Dart version is pinned here:
https://github.com/dependabot/dependabot-core/blob/6ca40e98b8d9b51950d28ed69f01012f563b5759/pub/Dockerfile#L9

This may also be relevant:
https://github.com/dependabot/dependabot-core/blob/6ca40e98b8d9b51950d28ed69f01012f563b5759/pub/helpers/pubspec.yaml#L1-L15

I'm not familiar enough with the Dart/Flutter ecosystem to know exactly what's happening... from the description it sounds like you might be using the beta `3.x` version of Flutter or Dart/pub??

**Comment by jeffwidman** [comment]
I wonder if this is also related?
* https://github.com/dependabot/dependabot-core/pull/6454

**Comment by JonasAlfredsson** [comment]
Hi, 

Are you sure you wanted to CC me? I have only produced one commit in this repository, and it was related to the github_actions folder :)

**Comment by sigurdm** [comment]
> The Dart version is pinned here:

The situation is a bit more complex. This is actually just the dart runtime used to run the dependency, services.

The actual version of the dependency services doing the resolution is decided here: https://github.com/dependabot/dependabot-core/blob/cafef062f5bb155e0735d6421a872f36d0240c19/pub/helpers/pubspec.yaml#L15

The version of dart flutter sdks  that is used when resolving is decided dynamically from the sdk-constraints by this script https://github.com/dependabot/dependabot-core/blob/2b60a710c985c470db70767c612a5c71602d853e/pub/helpers/bin/infer_sdk_versions.dart

**Comment by sigurdm** [comment]
> I wonder if this is also related? https://github.com/dependabot/dependabot-core/pull/6454

Yes, https://github.com/dependabot/dependabot-core/pull/6454 will (should) fix the issue of missing hashes.

**Comment by SiddheshKukade** [comment]
> This may also be relevant:
> 
> https://github.com/dependabot/dependabot-core/blob/6ca40e98b8d9b51950d28ed69f01012f563b5759/pub/helpers/pubspec.yaml#L1-L15
> 

Actually @jeffwidman I've tried to add these changes in the yaml but it didn't worked out for me.
> I'm not familiar enough with the Dart/Flutter ecosystem to know exactly what's happening... from the description it sounds like you might be using the beta `3.x` version of Flutter or Dart/pub??

Yes, it happened when we upgraded dart to 3.7.3
